### PR TITLE
[google.visualization] Add definition for Sankey

### LINKED ISSUE #50446: Missing type support for 'Sankey' in @types/google.visualization
- [x] I tried using the `@types/google.visualization` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @danludwig @gmoore-sjcorg @danmana @mlcheng @IvanBisultanov @glebm @shrujalshah28 @dckorben

## **Missing support for `Sankey` in index.d.ts**

`Sankey` exists on an an instance of google.visualization as per the Google Chart docs but is not exposed by the types for google.visualization.

### PR DESCRIPTION
**Pull request by martin-badin**
Closes: #50446

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/google.visualization/index.d.ts [hunk 1]
```diff
@@ -1724,6 +1731,61 @@ declare namespace google {
             yellowTo?: number;
         }
 
+        //#endregion
+        //#region Sankey
+
+        // https://developers.google.com/chart/interactive/docs/gallery/sankey
+        export class Sankey extends ChartBaseClearable {
+            draw(data: DataTable | DataView, options: SankeyChartOptions): void;
+            getBoundingBox(id: string): ChartBoundingBox;
```
Comment (danmana) [inline types/google.visualization/index.d.ts:1740]: The documentation is wrong here, the `getBoundingBox` does not actually exist at runtime

File: types/google.visualization/index.d.ts [hunk 2]
```diff
@@ -1724,6 +1731,61 @@ declare namespace google {
...
+            height?: number;
+            sankey?: {
+                iterations?: number;
+                link?: {
+                    color?: {
+                        fill?: string;
+                        fillOpacity?: number;
+                        stroke?: string;
+                        strokeWidth?: number;
+                    };
```
Comment (danmana) [inline types/google.visualization/index.d.ts:1756]:
```suggestion
                    color?: string | ChartStrokeFill;
```
Also supports a color string.

You will also have to add `fillOpacity` to `ChartFill`. I double-checked and it's also supported in other places (ex: chart backgroundColor)
```
export interface ChartFill {
            fill?: string;
            fillOpacity?: number;
        }
```

File: types/google.visualization/index.d.ts [hunk 3]
```diff
@@ -1724,6 +1731,61 @@ declare namespace google {
...
+                    colorMode?: 'unique';
+                    colors?: string[];
+                    interactivity?: boolean;
+                    label?: {
+                        bold?: boolean;
+                        color?: string;
+                        fontName?: string;
+                        fontSize?: number;
+                        italic?: boolean;
+                    };
```
Comment (danmana) [inline types/google.visualization/index.d.ts:1770]:
```suggestion
                    label?: ChartTextStyle;
```
You can reuse `ChartTextStyle`

File: types/google.visualization/index.d.ts [hunk 4]
```diff
@@ -1724,6 +1731,61 @@ declare namespace google {
...
+                    isHtml?: boolean;
+                    showColorCode?: true;
+                    textStyle?: {
+                        bold?: boolean;
+                        color?: string;
+                        fontName?: string;
+                        fontSize?: number;
+                        italic?: boolean;
+                    };
+                };
```
Comment (danmana) [inline types/google.visualization/index.d.ts:1785]:
```suggestion
                tooltip?: ChartTooltip;
```
You can reuse `ChartTooltip`

**Review by danmana — CHANGES_REQUESTED** [review]

**Comment by martin-badin** [comment]
Thanks, @danmana for your review! I added your suggestions.

**Review by danmana — APPROVED** [review]

**Comment by danmana** [comment]
Ready to merge

**Comment by danmana** [comment]
@martin-badin everything looks great 👍 
Thanks for your contribution!

**Pull request merged**